### PR TITLE
Add citation export from each page.

### PR DESCRIPTION
**Pull request by josephsnyder**
For now, a complicated string creation mess.

### CODE REVIEW

**Comment by josephsnyder** [comment]
Fixes #48

File: src/components/Citation.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  const populateAuthors = (publication) => {
+    const has_authors = publication.authors ? true : false;
+    if (has_authors) {
+      for (const author of authors) {
+        if (author.persona_firstname) {
+          descriptor_names.push(author.persona_lastname);
+          full_names.push(`${author.persona_firstname[0]}. ${author.persona_lastname}`)
+          last_names.push(author.persona_lastname)
+        } else {
+            descriptor_names.push(author.author_fullname);
```
Comment (thewtex) [inline src/components/Citation.js:25]: We should use `author_fullname` first, if available. Persona support was added later.

File: src/templates/publication.js [hunk 2]
```diff
@@ -107,6 +108,7 @@ const Render = ({ data }) => {
 	<center>{coverImage}</center>
 	<br/>
         {citeLink}
+    <Citation publication={publication}/>
```
Comment (thewtex) [inline src/templates/publication.js:111]: Can this go after the abstract, on one line, *Export citation: <dropdown> EXPORT*?

**Review by thewtex — CHANGES_REQUESTED** [review]
@josephsnyder nice!!

A few items noted when testing: 

- The BibTex file extension should be *.bib*.
- The BibTex article identifier cannot have spaces.
-  BibTex values need quotes around strings.
- Text does not have spacing correct, should use abbreviations for the first names, quotes around title, handle the last author correctly (there is a trailing *and*). End in a period.
- XML has an extra`",` after `</Location>`.

File: src/templates/publication.js (hunk 2 above)
Comment (josephsnyder) [reply src/templates/publication.js:111]: Done

File: src/components/Citation.js (hunk 1 above)
Comment (josephsnyder) [reply src/components/Citation.js]: I reworked most of the names strings so they should be more correct.  

According to the recent exports from the Midas Journal, the text citations don't have any quotes in them but all of the others have been updated

Comment (thewtex) [reply src/components/Citation.js]: Looking better.

Remaining issues:

- *BibTex* -> *BibTeX*.
- The file name for the bib should be something like *publication.bib*.
- There is still an wayward `"` in the XML.
- The Text citations should have quotes around the title (if this was wrong in the Midas Journal, it should still be corrected).
- The Text citations should have the URL and journal name.

Comment (josephsnyder) [reply src/components/Citation.js]: How does this look?

```
Kim E., Johnson H., Williams N. "Affine Transformation for Landmark Based Registration Initializer in ITK". The MIDAS Journal. 2011 Jul. http://hdl.handle.net/10380/3299
```

Comment (thewtex) [reply src/components/Citation.js]: Excellent!

**Comment by thewtex** [comment]
In addition to the review items, a rebase is required to address conflicts.

**Review by thewtex — APPROVED** [review]

**Pull request merged**